index.rst

```diff
@@ -41,40 +41,38 @@ with version control systems.
 File Format
 ===========
 
-EditorConfig files use an INI format that is compatible with the format used
-by `Python configparser Library`_. In an EditorConfig file (usually named
-``.editorconfig``), all beginning whitespace in each line is ignored. Each
-line must be one of the following:
-
-- Empty Line: An empty string.
-- Comment: A line starting with a ``;`` or a ``#``.
-- Section Title: A line that starts with a ``[``, and that ends with a ``]``.
-- Assignment: A line separated by an `=` into two parts.
-
-Any line that is not one of the above is invalid. "Inline comment" in
-appearance, i.e., a line that has a ``;`` or a ``#`` in the middle, is
-undefined. For convenience, we also define the following terminologies:
-
-- Section Name: The string between the beginning ``[`` and the ending ``]`` in
-  a Section Title.
-- Section: The lines starting from a Section Title until the beginning of the
-  next Section Title or end of file.
-- Property: The part before the first `=` in an Assignment.
-- Value: The part after the first `=` in an Assignment.
-
-In EditorConfig files, the Section Names are filepath globs, similar to the
-format accepted by gitignore. ``[`` and ``]`` are allowed in the Section
-Names. Forward slashes (``/``) are used as path separators and Backslashes
-(``\\``) are not allowed as path separators (even on Windows). Comments should
-always go on individual lines. EditorConfig files should be UTF-8 encoded, with
-either CRLF or LF line separators.
+EditorConfig uses an INI file format. In an EditorConfig file (usually named
+``.editorconfig``), all beginning whitespace on each line is considered
+irrelevant. Each line must be one of the following:
+
+- Blank: contains only whitespace characters.
+- Comment: starts with a ``;`` or a ``#``.
+   - Inline comments are not supported.
+- Section Header: starts with a ``[`` and ends with a ``]``.
+   - May not use any non-whitespace characters outside of the surrounding
+     brackets.
+   - May contain any characters between the square brackets (e.g.,
+     ``[`` and ``]`` are allowed).
+   - Forward slashes (``/``) are used as path separators.
+   - Backslashes (``\\``) are not allowed as path separators (even on Windows).
+   - Section Name: the string between the beginning ``[`` and the ending ``]``.
+   - Section: the lines starting from a Section Header until the beginning of the
+     next Section Header or the end of the file.
+- Property: contains a key-value pair, separated by an `=`.
```

Comment:
Renamed this from Assignment to Property for consistency with the rest of the document as well as the Wikipedia article about the INI file format.

Reply:
From Wikipedia, Does "Property" mean the left-hand side of the key-value pair, i.e., the key?

Reply:
No, the Wikipedia article always refers to the property as the whole thing. Unfortunately, it uses key and property interchangeably for the same thing, which is not ideal.

The YML specification refers to them as [key: value pairs](https://yaml.org/spec/1.2/spec.html#key:%20value%20pair//) and so does JSON, JavaScript and many other languages.

Hmm... it looks like CSS refers to each [property declaration as a name and value](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty).

I'm actually OK renaming "key" back to "name" if that's more clear. It's just not a consistent nomenclature throughout languages, unfortunately.

Reply:
I see [CSS spec](https://www.w3.org/TR/2019/CR-css-syntax-3-20190716/) uses "property" loosely to refer to either the left-hand side or the whole assignment. "Property" seems to me is sort of loose. Given that "property" is used loosely else where, I'm not sure it's worth to abandon the current terminology here; we probably have a lot of wiki pages and there are other online resources that have already followed the current definition.

Reply:
[Yaml spec](https://yaml.org/spec/1.2/spec.html) also seems like so:

>  Both the node’s properties and node content are optional.

Reply:
That's not being referred to as the left side. Where do you see it being referred to as the left side? I don't think Property is loose at all. It _always_ refers to the whole thing, where I've seen it used. When the CSS spec says "color property" don't confuse that to mean that "color" is itself a property. The (color) property is a property with a key named color and a hex value. It's still a property, but they are using the key of the property as a sort of label so you know the property to which they are referring.

Reply:
My name is Jed Mao. If you saw 100 names in a file, you might say, "which line is Jed's name on?" Same thing. Which line is the color's property on?

Reply:
Here's the relevant quote from a highlighted piece of text from the above CSS spec doc:

> CSS renderers must treat as invalid (and ignore as appropriate) any at-rules, properties, property values, keywords, and other syntactic constructs for which they have no usable level of support.

I don't see a clear meaning of "property" in its other occurrences in that doc, but in this sentence, I think "properties" more likely means the left-hand side than the whole thing.

Reply:
I looked up [JSON spec](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf) as well, and there is no mention of "property." Right now I've had no luck in searching yaml, CSS, and JSON specs for a clear definition of "property" as used in a similar way as we desire to use. If the word "property" is so vague and others are avoiding using it, I think it might be easier to just avoid using this word, and let existing online resources refer to whatever it used to mean, maybe loosely, maybe more precisely. Alternatively we can use word like "declaration" for key-value pair, "key" for the left-hand side, "value" for the right-hand side. What do you think?

Reply:
I’m OK with that. I got “declaration” [from PostCSS](https://github.com/postcss/postcss/blob/master/lib/declaration.es6).

Reply:
OK. Shall we go for "declaration"? (BTW "statement" might also be a good candidate)

Reply:
Yeah. I think we should stick with declaration. It also seems to be consistent with [the description of the const declaration in JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#description).